### PR TITLE
Upgrade minting mechanism for accurate minting at the tx level

### LINKED ISSUE #892: Upgrade the minting mechanism to mint at the tx/handler level
**Describe the feature**
Reward the relayers and challenge senders at the msg handler level, and add the remainder to the fee collector to be `properly` (currently broken) distributed to the DAO and the Proposer.

On blockreward, take the bigger proportion and apply a FLOOR function to it and pay the actor.
Send the remainder of the feescollected to the actor with the smaller proportion.

### PR DESCRIPTION
**Pull request by andrewnguyen22**
closes #892

### CODE REVIEW

File: x/nodes/keeper/params.go [hunk 1]
```diff
@@ -90,10 +90,20 @@ func (k Keeper) RelaysToTokensMultiplier(ctx sdk.Ctx) sdk.Int {
 	return sdk.NewInt(1000) // todo parameterize
 }
 
-func (k Keeper) NodeCutOfReward(ctx sdk.Ctx) sdk.Int {
-	daoAllocation := k.DAOAllocation(ctx)
-	proposerAllocation := k.ProposerAllocation(ctx)
-	return sdk.NewInt(100 - daoAllocation - proposerAllocation)
+func (k Keeper) NodeCutOfReward(ctx sdk.Ctx, reward sdk.Int) (toNode sdk.Int, toFeeCollector sdk.Int) {
```
Comment (luyzdeleon) [inline x/nodes/keeper/params.go]: NodeReward would be a better name?

Comment (luyzdeleon) [reply x/nodes/keeper/params.go]: Also `nodeReward` and `feesCollected` or `collectedFees` or `fees` would make sense for the return values.

File: x/nodes/keeper/reward.go [hunk 2]
```diff
@@ -4,147 +4,48 @@ import (
...
 
 // award coins to an address (will be called at the beginning of the next block)
 func (k Keeper) RewardForRelays(ctx sdk.Ctx, relays sdk.Int, address sdk.Address) {
-	award, _ := k.getValidatorAward(ctx, address)
 	coins := k.RelaysToTokensMultiplier(ctx).Mul(relays)
-	k.setValidatorAward(ctx, award.Add(coins), address)
-	ctx.Logger().Info("Custom award of " + coins.String() + " set for " + address.String())
+	toNode, toFeeCollector := k.NodeCutOfReward(ctx, coins)
+	k.mint(ctx, toNode, address)
+	k.mint(ctx, toFeeCollector, k.getFeePool(ctx).GetAddress())
```
Comment (luyzdeleon) [inline x/nodes/keeper/reward.go]: Check if feeCollector values are 0 as they are params and could be zero'd out by governance.

File: x/nodes/keeper/reward.go [hunk 3]
```diff
@@ -4,147 +4,48 @@ import (
...
-
-func (k Keeper) getValidatorAward(ctx sdk.Ctx, address sdk.Address) (coins sdk.Int, found bool) {
-	store := ctx.KVStore(k.storeKey)
-	value := store.Get(types.KeyForValidatorAward(address))
-	if value == nil {
-		return sdk.ZeroInt(), false
+	// get the dao and proposer % ex DAO .1 or 10% Proposer .01 or 1%
+	daoAllocation := sdk.NewDec(k.DAOAllocation(ctx))
+	proposerAllocation := sdk.NewDec(k.ProposerAllocation(ctx))
+	totalAllocationWithoutNodeCut := daoAllocation.Add(proposerAllocation)
```
Comment (luyzdeleon) [inline x/nodes/keeper/reward.go]: Would be less confusing to call this variable `daoAndProposerAllocation` or `totalAllocation`

**Review by luyzdeleon — CHANGES_REQUESTED** [review]
Can you pinpoint to where the fees collector is zeroed out to make sure it doesn't leave "spare change"?

File: x/nodes/keeper/params.go (hunk 1 above)
Comment (andrewnguyen22) [reply x/nodes/keeper/params.go]: Will fix!

File: x/nodes/keeper/reward.go (hunk 2 above)
Comment (andrewnguyen22) [reply x/nodes/keeper/reward.go]: Well, should work either way, but will add for sanity

File: x/nodes/keeper/reward.go (hunk 3 above)
Comment (andrewnguyen22) [reply x/nodes/keeper/reward.go]: Will fix

**Pull request merged**